### ISSUE TITLE
Mirth does not start up on Apple M1

### ISSUE THREAD

**Issue by jas550** [nextgenhealthcare/connect-docker#15]
Mirth does not start up on Apple M1. It either throws the following error or just hangs:

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x0000004017c7a324, pid=1202, tid=1759
#
# JRE version: OpenJDK Runtime Environment 18.9 (11.0.8+10) (build 11.0.8+10)
# Java VM: OpenJDK 64-Bit Server VM 18.9 (11.0.8+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# J 459 c1 java.util.zip.ZipFile.ensureOpen()V java.base@11.0.8 (40 bytes) @ 0x0000004017c7a324 [0x0000004017c7a300+0x0000000000000024]
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /opt/connect/hs_err_pid1202.log
Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#
qemu: uncaught target signal 6 (Aborted) - core dumped
```

It will be useful to build an arm64 specific docker container that can be used on Apple M1 machines.

**Comment by cturczynskyj** [comment]
There's some discussion about this topic in the [Connect GitHub repo](https://github.com/nextgenhealthcare/connect/discussions/4984). It sounds like the issue may be with MacOS Monterey and not the M1 based machines themselves but that hasn't been confirmed yet.

**Comment by jas550** [comment]
I've tested the mirth docker container on amd64 Mac OS Monterey and the container works fine. However, the mirth client crashes. Therefore I think the discussion only applies when running mirth or mirth client on the host machine.

**Comment by jas550** [comment]
@cturczynskyj Why was this closed. This is still an issue. Mirth docker container still doesn't work on Mac M1 machines. The mirth client also crashes on Mac OS Monterey amd64.

**Comment by cturczynskyj** [comment]
Maybe I misunderstood your comment. It sounded like you were saying that it was only an issue with the client which wouldn't be Docker related. Are you saying that the Docker container fails to start on your Monterey machine?

**Comment by jas550** [comment]
@cturczynskyj I was replying back to your earlier comment (below). 

> There's some discussion about this topic in the [Connect GitHub repo](https://github.com/nextgenhealthcare/connect/discussions/4984). It sounds like the issue may be with MacOS Monterey and not the M1 based machines themselves but that hasn't been confirmed yet.

I've created a table to specify what works in which environment:

|Test|Environment|Works|
|---|---|---|
|Mirth Client| amd64 Mac OS Monterey| No
|Mirth Client| arm64 (m1) Mac OS Monterey|No
|Mirth docker container|amd64 Mac OS Monterey| Yes
|Mirth docker container| arm64 (m1) Mac OS Monterey|No

Let me know if you require any further information to progress this issue.

**Comment by jas550** [comment]
@cturczynskyj Can you please open this issue as it wasn't resolved. Or do you want me to open a new issue?

**Comment by pladesma** [comment]
Related issue https://github.com/nextgenhealthcare/connect-docker/issues/24

**Comment by jas550** [comment]
@pladesma @lmillergithub Do you know when this issue will be resolved?

**Comment by svenczbg** [comment]
Did you try to enable "Use Rosetta for x86/amd64 emulation in Apple Silicon" option in docker desktop? You can find that in the settings tab "Features in development".
That allowed me to run the amd64 image on my M1 Mac.
